src/components/autosuggest/_macro.spec.js

```diff
@@ -5,130 +5,177 @@ import * as cheerio from 'cheerio';
-                isEditable: false,
-            }),
-        );
+    describe('GIVEN: Params: isEditable', () => {
+        describe('WHEN: isEditable is false', () => {
+            const $ = cheerio.load(
+                renderComponent('autosuggest', {
+                    ...EXAMPLE_AUTOSUGGEST,
+                    isEditable: false,
+                }),
+            );
 
-        expect($('.ons-autosuggest').hasClass('ons-js-address-not-editable')).toBe(true);
+            test('THEN: it has the ons-js-address-not-editable class to indicate that component is not editable', () => {
+                expect($('.ons-autosuggest').hasClass('ons-js-address-not-editable')).toBe(true);
+            });
+        });
     });
 
-    it('has special class to indicate that component input is mandatory', () => {
-        const $ = cheerio.load(
-            renderComponent('autosuggest', {
-                ...EXAMPLE_AUTOSUGGEST,
-                mandatory: true,
-            }),
-        );
+    describe('GIVEN: Params: mandatory', () => {
+        describe('WHEN: mandatory is true', () => {
+            const $ = cheerio.load(
+                renderComponent('autosuggest', {
+                    ...EXAMPLE_AUTOSUGGEST,
+                    mandatory: true,
+                }),
+            );
 
-        expect($('.ons-autosuggest').hasClass('ons-js-address-mandatory')).toBe(true);
+            test('THEN: it has the ons-js-address-mandatory class to indicate that component input is mandatory', () => {
+                expect($('.ons-autosuggest').hasClass('ons-js-address-mandatory')).toBe(true);
+            });
+        });
     });
 
-    it('has additionally provided container style classes', () => {
-        const $ = cheerio.load(
-            renderComponent('autosuggest', {
-                ...EXAMPLE_AUTOSUGGEST,
-                containerClasses: 'extra-class another-extra-class',
-            }),
-        );
+    describe('GIVEN: Params: containerClasses', () => {
+        describe('WHEN: containerClasses is provided', () => {
+            const $ = cheerio.load(
+                renderComponent('autosuggest', {
+                    ...EXAMPLE_AUTOSUGGEST,
+                    containerClasses: 'extra-class another-extra-class',
+                }),
+            );
 
-        expect($('.ons-autosuggest').hasClass('extra-class')).toBe(true);
-        expect($('.ons-autosuggest').hasClass('another-extra-class')).toBe(true);
+            test('THEN: it has additionally provided container style classes', () => {
+                expect($('.ons-autosuggest').hasClass('extra-class')).toBe(true);
+                expect($('.ons-autosuggest').hasClass('another-extra-class')).toBe(true);
+            });
+        });
     });
 
-    describe('input', () => {
-        it('uses the `input` component with the expected parameters', () => {
+    describe('GIVEN: Params: input', () => {
+        describe('WHEN: input param is provided', () => {
```

Comment:
The example for this test is that long that it would probably be better if it was in the examples file and imported